client/src/components/Task/PlayPauseButton/PlayPauseButton.js

```diff
@@ -0,0 +1,101 @@
+import "./PlayPauseButton.modules.scss";
+import React from "react";
+import IconButton from "@material-ui/core/IconButton";
+import PauseIcon from '@material-ui/icons/Pause';
+import PlayArrowIcon from '@material-ui/icons/PlayArrow';
+import Box from "@material-ui/core/Box";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import Fade from "@material-ui/core/Fade";
+
+export default function PlayPauseButton(props) {
+    const [started, setStarted] = React.useState(false);
+    const [progress, setProgress] = React.useState(0.0);
+    const [timerID, setTimerID] = React.useState(0);
+    const {maxTime} = props;
+
+    //Callback when play pause button is pressed
+    function playClick() {
+        setStarted((prev) => {
+            let newVal = !prev;
+
+            //If now stopped, clear interval
+            if(!newVal)
+                clearInterval(timerID);
+
+            return newVal;
+        });
+
+        //If progress was 100, then pause was pressed, then reset progress
+        if(progress === 100)
+            setProgress(0);
+    }
+
+    //Start interval once counter is started
+    React.useEffect(() => {
+        if(started)
+        {
+            setTimerID(setInterval(() => {
+                setProgress((prevProgress) => {
+                    let newProgress = prevProgress + 1 / maxTime * 100.0;
+                    if(newProgress > 100)
+                        newProgress = 100;
+                    else if(newProgress < 0)
+                        newProgress = 0;
+
+                    return newProgress;
+                });
+            }, 1000));
+        }
+    }, [started, maxTime]);
+
+    //Effect to clear timer when progress is done
+    React.useEffect(() => {
+        if(progress === 100)
+        {
+            clearInterval(timerID);
+            setStarted((prev) => {
```

Comment:
same here, prev/ newVal doesn't add any extra info, can consider

setStarted(!started) 

or
```
setStarted(prevStartStatus => {
 let newStartStatus = !prevStartStatus
 return newStartStatus
}
```

Reply:
After removing the redundant code, I remember why I did it like that. if I use setStarted(!started) then React will complain that "started" is not set as dependency of useEffect

If it is a dependency, then infinite loop will happen because setState is inside a useEffect which has the state being set as dependency

Reply:
^ added comments in code explaining why

Reply:
oww no wonder, I was curious also haha, okay nice!